google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -484,7 +484,7 @@ def execute_query sql, params: nil, types: nil, single_use: nil,
           request_options = Convert.to_request_options request_options,
                                                        tag_type: :request_tag
           single_use_tx = single_use_transaction single_use
-          route_to_leader = LARHeaders.execute_query true
+          route_to_leader = LARHeaders.execute_query !single_use_tx
```

Comment:
I am pretty sure it should be just `false`. If it's single-use (as specified by the caller) it's RO explicitly; if  `nil` is specified by the caller, the default case for `execute_streaming_sql` is a "temporary read-only transaction with strong concurrency" https://github.com/googleapis/googleapis/blob/53af3b727f84acc34e31c23f3b6e7b8aa4b7e837/google/spanner/v1/spanner.proto#L716-L717